docs/DeveloperGuide.md

```diff
@@ -1,34 +1,234 @@
+
+![modelcomponet](DG_Diagrams/ModelComponent.png)
+
+#### 2.1.4. Storage Component
+The Storage component is responsible for the following tasks:
+- Stores the favourite commands in a file in a certain format and retrieves this information everytime 
+the user enters the application.
+- Stores the number of times each bus stop has been typed by the user since he/she started using the application and 
+retrieves it everytime the user enters the application.
+    + This information is used to find out the user's most frequently searched location so far and display it to the 
+    user. 
+- These files are updated after every execution of a user command.
+
+The following class diagram briefly explains how different classes in the Storage component interact with each other.
+
+![StorageComponent](DG_Diagrams/StorageComponent.png)
+
+### 2.2. Implementation
+This section provides details of how the main features of Nav@NUS have been implemented.
+
+#### 2.2.1 Direct Route Finder (`/route` Feature)
+
+`/route <location1> /to <location2>` is the command that has to entered by the user to see all direct bus routes 
+available from **location1** to **location2**.
+
+The class diagram in the figure below shows how different classes used for implementation of the `/route` command 
+are linked to each other. 
+
+![RouteCommandClass](DG_Diagrams/RouteCommandClass.png)
+
+The `RouteCommand#executeCommand()` method of RouteCommand Class executes the command in the following steps:
+1. Calls `RouteParser#getLocations()` to get the locations entered by the user in the order of starting location and 
+destination.
+    - The `RouteParser#getLocations()` method throws an exception if the locations or the delimiter `/to` is missing.
+2. Calls `RouteCommand#checkLocations()` to make sure location strings are not empty or the same.
+    - The `RouteCommand#checkLocation()` method throws an exception if locations are empty or the same.
+3. Calls `RouteCommand#similarityCheck()` to check if the locations are not in the list of bus stops but 
+are similar.
+    - The `RouteCommand#similarityCheck()` method calls the static `SimilarityCheck#similarLoc()` method and returns 
+    a list of similar locations, if any.
+    - If the list of similar locations is empty, `RouteCommand#executeCommand()` performs step 4 given below, 
+    else it calls the static method, `Ui#printPossibleLocsMessage()`, to print the list of similar locations. 
+4. Calls static `BusData#possibleBuses()` to get a list of buses with their routes from the starting location to 
+the destination.
+   - `BusData#possibleBuses()` calls `Bus#getPossibleRoute()` to check for a possible route for the given bus number.
+   - `BusData#possibleBuses()` repeats this call for all bus numbers.
+
+The following sequence diagram explains the above steps when the user enters `/route loc1 /to loc2`.
+
+![Overview](DG_Diagrams/RouteCommand.png)
```

Comment:
I'm not sure if its because the screen resolution is not good but the RouteParser : is missing?